src/freenas/etc/ix.rc.d/ix-shutdown

```diff
@@ -11,25 +11,7 @@
 
 do_shutdown()
 {
-	local IFS="|"
-	local f="ini_type ini_command ini_script ini_when"
-	eval local $f
-	local sf=$(var_to_sf $f)
-
-	RO_FREENAS_CONFIG=$(ro_sqlite ${name} 2> /tmp/${name}.fail && rm /tmp/${name}.fail)
-	trap 'rm -f ${RO_FREENAS_CONFIG}' EXIT
-
-	${FREENAS_SQLITE_CMD} ${RO_FREENAS_CONFIG} \
-	"SELECT $sf FROM tasks_initshutdown WHERE ini_when = 'shutdown' AND ini_enabled = 1 ORDER BY id" | \
-	while eval read -r $f; do
-		if [ "${ini_type}" = "command" ]; then
-			eval ${ini_command}
-		else
-			if [ -e "${ini_script}" ]; then
-				sh -c "exec ${ini_script}"
-			fi
-		fi
-	done
+	/usr/local/bin/midclt call initshutdownscript.execute_init_tasks SHUTDOWN > /dev/null
```

Comment:
This will break if shutdown scripts take more than midclt timeout.

Reply:
That is indeed true. @william-gr do you think it would be better if we perhaps made this method a job and waited for it here ? 🤔

Reply:
We may not want to wait forever which would prevent shutdown due to use script error but the default of 60 seconds may be too low.

Perhaps try to increase it as a first step? Something like 5 or 10 min?

Reply:
Agreed. I think it would be better as well if we reflected this in the docs to prep the user that we would be waiting this much time at a maximum for the shutdown tasks. I will have a PR up for you for shutdown, pre-init and post-init asap. Thank you for pointing this out :)